artifact/scripter.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package artifact
+
+import (
+	"path/filepath"
+	"regexp"
+
+	"github.com/pkg/errors"
+)
+
+type Scripts struct {
+	names []string
+}
+
+var availableScriptType = map[string]bool{
+	"Idle":                true,
+	"Sync":                true,
+	"ArtifactDownload":    true,
+	"ArtifactPreinstall":  true,
+	"ArtifactInstall":     true,
+	"Reboot":              true,
+	"ArtifactPostinstall": true,
+	"ArtifactCommit":      true,
```

Comment:
Thanks for spotting this. Probably I took a look at some subset `Order that states will usually be executed in:`. Anyway, fixed now and missing states are added.